src/common/AmazonS3Backend.ts

```diff
@@ -276,27 +276,105 @@ export class AmazonS3Backend implements IBackend {
    * }
    */
   public uploadFile(params: any): Promise<any> {
-    // add ManagedUpload object to the file
-    params.Body.s3upload = new AWS.S3.ManagedUpload({
-      params: params,
-      service: this.s3,
-      partSize: 1024 * 1024 * 10,
-      queueSize: 1
-    });
-    params.Body.s3upload.computeChecksums = true;
 
-    // add event listener to update upload progress
-    params.Body.s3upload.on('httpUploadProgress', function(progress: any) {
-      if (progress.total) {
-        let percent = (progress.loaded * 100) / progress.total;
-        params.Dropzone.emit('uploadprogress', params.Body, percent, progress.loaded);
-      }
-    });
+    let completedParts = [];
+    let s3client = this.s3;
 
     return new Promise((resolve: any, reject: any) => {
-      params.Body.s3upload.send(function(err: any, data: any) {
-        resolve({err: err, data: data});
-      });
+      // callback function to be used with s3.listParts()
+      // creates a ManagedUpload from an existing multipart upload.
+      let resumeUploadCallback = function(err: any, data: any) {
+        if (err) {
+          console.log(err);
+          resolve({err: err, data: data});
+          return;
+        }
+
+        // collect data for parts that are already uploaded to aws.
+        let byteCount = 0;
+        for (let i of Object.keys(data.Parts)) {
+          let partNumber = data.Parts[i].PartNumber;
+          completedParts[partNumber] = {ETag: data.Parts[i].ETag, PartNumber: partNumber};
+          byteCount = byteCount + data.Parts[i].Size;
+        }
+
+        let upload = new AWS.S3.ManagedUpload({
+          params: params,
+          service: s3client,
+          partSize: 1024 * 1024 * 10,
+          queueSize: 1
```

Comment:
Just a thought, maybe we should define the ManagedUpload config as a variable outside this function since we create ManagedUploads in the resume callback and findMultipart callback. That way if we need to modify it, we can do so in one place, instead of two.

Reply:
Setting this in one place sounds like a good idea to me.